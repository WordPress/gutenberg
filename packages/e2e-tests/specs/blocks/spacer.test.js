/**
 * WordPress dependencies
 */
import {
	clickBlockAppender,
	getEditedPostContent,
	createNewPost,
	dragAndResize,
} from '@wordpress/e2e-test-utils';

describe( 'Spacer', () => {
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'can be created by typing "/spacer"', async () => {
		// Create a spacer with the slash block shortcut.
		await clickBlockAppender();
		await page.keyboard.type( '/spacer' );
		await page.keyboard.press( 'Enter' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be resized using the drag handle and remains selected after being dragged', async () => {
		// Create a spacer with the slash block shortcut.
		await clickBlockAppender();
		await page.keyboard.type( '/spacer' );
		await page.keyboard.press( 'Enter' );

		const resizableHandle = await page.$( '.block-library-spacer__resize-container .components-resizable-box__handle' );
		await dragAndResize( resizableHandle, { x: 0, y: 50 } );
		expect( await getEditedPostContent() ).toMatchSnapshot();

		const selectedSpacer = await page.$( '[data-type="core/spacer"].is-selected' );
		expect( selectedSpacer ).not.toBe( null );
	} );
} );
