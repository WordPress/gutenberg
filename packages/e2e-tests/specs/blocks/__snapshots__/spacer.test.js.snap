// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer can be created by typing "/spacer" 1`] = `
"<!-- wp:spacer -->
<div style=\\"height:100px\\" aria-hidden=\\"true\\" class=\\"wp-block-spacer\\"></div>
<!-- /wp:spacer -->"
`;

exports[`Spacer can be resized using the drag handle and remains selected after being dragged 1`] = `
"<!-- wp:spacer {\\"height\\":150} -->
<div style=\\"height:150px\\" aria-hidden=\\"true\\" class=\\"wp-block-spacer\\"></div>
<!-- /wp:spacer -->"
`;
